File: frontend/src/types/Progress.ts
export interface ScoreTrendPoint {
  quiz_id: number
  completed_at: string | null
  score_percentage: number
}

export interface TopicAccuracy {
  topic: string
  accuracy_percentage: number
  attempts: number
}

export interface ProgressSummary {
  total_quizzes: number
  average_score_percentage: number
  score_trend: ScoreTrendPoint[]
  topic_accuracy: TopicAccuracy[]
}

File: backend/app/schemas/progress.py
from datetime import datetime

from pydantic import BaseModel


class ScoreTrendPoint(BaseModel):
    quiz_id: int
    completed_at: datetime | None
    score_percentage: float


class TopicAccuracy(BaseModel):
    topic: str
    accuracy_percentage: float
    attempts: int


class ProgressSummary(BaseModel):
    total_quizzes: int
    average_score_percentage: float
    score_trend: list[ScoreTrendPoint]
    topic_accuracy: list[TopicAccuracy]
